`useless-break` flags necessary `break` in `case`

**Describe the bug**
`useless-break` flags a `break` statement within a `case`, followed by a label

**To Reproduce**
Steps to reproduce the behavior:
On version `v1.7.0` (8ece20b0789c517bd3a6742db0daa4dd5928146d)

```shell
revive --config revive.toml demo/...
```

```toml
[rule.useless-break]
Enabled = true
```

```go
// demo/main/main.go
package main

import "fmt"

func main() {
	switch 1 {
	case 1:
		fmt.Println("foo bar")
		if false { // Add code that jumps to the label so it compiles.
			goto SOME_LABEL
		}
		break
	SOME_LABEL:
		panic(nil)
	}
}
```

**Expected behavior**
No lint error, since the `break` prevents the control flow from entering `SOME_LABEL`.

**Logs**
`demo/main/main.go:12:3: useless break in case clause`

**Desktop (please complete the following information):**
 - OS: `openSUSE Leap 15.6`
 - `go1.24.1 linux/amd64`

--- a/rule/useless_break.go
+++ b/rule/useless_break.go
@@ -63,7 +63,7 @@ func (w *lintUselessBreak) inspectCaseStatement(body []ast.Stmt) {
 	}
 
 	s := body[l-1] // pick the last statement
-	if !isUnlabeledBreak(s) {
+	if !isUnlabelledBreak(s) {
 		return
 	}
 
@@ -79,7 +79,7 @@ func (w *lintUselessBreak) inspectCaseStatement(body []ast.Stmt) {
 	})
 }
 
-func isUnlabeledBreak(stmt ast.Stmt) bool {
+func isUnlabelledBreak(stmt ast.Stmt) bool {
 	s, ok := stmt.(*ast.BranchStmt)
 	return ok && s.Tok == token.BREAK && s.Label == nil
 }

--- a/rule/useless_break.go
+++ b/rule/useless_break.go
@@ -47,21 +47,21 @@ func (w *lintUselessBreak) Visit(node ast.Node) ast.Visitor {
 		w.inLoopBody = false
 		return nil
 	case *ast.CommClause:
-		for _, n := range v.Body {
-			w.inspectCaseStatement(n)
+		for i, n := range v.Body {
+			w.inspectCaseStatement(n, v.Body[i:])
 		}
 		return nil
 	case *ast.CaseClause:
-		for _, n := range v.Body {
-			w.inspectCaseStatement(n)
+		for i, n := range v.Body {
+			w.inspectCaseStatement(n, v.Body[i:])
 		}
 		return nil
 	}
 	return w
 }
 
-func (w *lintUselessBreak) inspectCaseStatement(n ast.Stmt) {
-	switch s := n.(type) {
+func (w *lintUselessBreak) inspectCaseStatement(s ast.Stmt, followings []ast.Stmt) {
+	switch s := s.(type) {
 	case *ast.BranchStmt:
 		if s.Tok != token.BREAK {
 			return // not a break statement
@@ -73,6 +73,13 @@ func (w *lintUselessBreak) inspectCaseStatement(n ast.Stmt) {
 		if w.inLoopBody {
 			msg += " (WARN: this break statement affects this switch or select statement and not the loop enclosing it)"
 		}
+
+		for _, f := range followings {
+			if _, ok := f.(*ast.LabeledStmt); ok {
+				return // there is a label statement following the break statement
+			}
+		}
+
 		w.onFailure(lint.Failure{
 			Confidence: 1,
 			Node:       s,

--- a/rule/useless_break.go
+++ b/rule/useless_break.go
@@ -47,20 +47,20 @@ func (w *lintUselessBreak) Visit(node ast.Node) ast.Visitor {
 		w.inLoopBody = false
 		return nil
 	case *ast.CommClause:
-		for i, n := range v.Body {
-			w.inspectCaseStatement(n, v.Body[i:])
+		if l := len(v.Body); l > 0 {
+			w.inspectCaseStatement(v.Body[l-1])
 		}
 		return nil
 	case *ast.CaseClause:
-		for i, n := range v.Body {
-			w.inspectCaseStatement(n, v.Body[i:])
+		if l := len(v.Body); l > 0 {
+			w.inspectCaseStatement(v.Body[l-1])
 		}
 		return nil
 	}
 	return w
 }
 
-func (w *lintUselessBreak) inspectCaseStatement(s ast.Stmt, followings []ast.Stmt) {
+func (w *lintUselessBreak) inspectCaseStatement(s ast.Stmt) {
 	switch s := s.(type) {
 	case *ast.BranchStmt:
 		if s.Tok != token.BREAK {
@@ -72,12 +72,6 @@ func (w *lintUselessBreak) inspectCaseStatement(s ast.Stmt, followings []ast.Stm
 		msg := "useless break in case clause"
 		if w.inLoopBody {
 			msg += " (WARN: this break statement affects this switch or select statement and not the loop enclosing it)"
-		}
-
-		for _, f := range followings {
-			if _, ok := f.(*ast.LabeledStmt); ok {
-				return // there is a label statement following the break statement
-			}
 		}
 
 		w.onFailure(lint.Failure{

--- a/rule/useless_break.go
+++ b/rule/useless_break.go
@@ -47,37 +47,39 @@ func (w *lintUselessBreak) Visit(node ast.Node) ast.Visitor {
 		w.inLoopBody = false
 		return nil
 	case *ast.CommClause:
-		if l := len(v.Body); l > 0 {
-			w.inspectCaseStatement(v.Body[l-1])
-		}
+		w.inspectCaseStatement(v.Body)
 		return nil
 	case *ast.CaseClause:
-		if l := len(v.Body); l > 0 {
-			w.inspectCaseStatement(v.Body[l-1])
-		}
+		w.inspectCaseStatement(v.Body)
 		return nil
 	}
 	return w
 }
 
-func (w *lintUselessBreak) inspectCaseStatement(s ast.Stmt) {
-	switch s := s.(type) {
-	case *ast.BranchStmt:
-		if s.Tok != token.BREAK {
-			return // not a break statement
-		}
-		if s.Label != nil {
-			return // labeled break statement, usually affects a nesting loop
-		}
-		msg := "useless break in case clause"
-		if w.inLoopBody {
-			msg += " (WARN: this break statement affects this switch or select statement and not the loop enclosing it)"
-		}
-
-		w.onFailure(lint.Failure{
-			Confidence: 1,
-			Node:       s,
-			Failure:    msg,
-		})
+func (w *lintUselessBreak) inspectCaseStatement(body []ast.Stmt) {
+	l := len(body)
+	if l == 0 {
+		return // empty body, nothing to do
 	}
+
+	s := body[l-1] // pick the last statement
+	if !isUnlabeledBreak(s) {
+		return
+	}
+
+	msg := "useless break in case clause"
+	if w.inLoopBody {
+		msg += " (WARN: this break statement affects this switch or select statement and not the loop enclosing it)"
+	}
+
+	w.onFailure(lint.Failure{
+		Confidence: 1,
+		Node:       s,
+		Failure:    msg,
+	})
+}
+
+func isUnlabeledBreak(stmt ast.Stmt) bool {
+	s, ok := stmt.(*ast.BranchStmt)
+	return ok && s.Tok == token.BREAK && s.Label == nil
 }

--- a/testdata/useless_break.go
+++ b/testdata/useless_break.go
@@ -87,4 +87,25 @@ func UselessBreaks() {
 	default:
 		s.errorf("range can't iterate over %v", val)
 	}
+
+	// issue #1281
+	switch 1 {
+	case 1:
+		fmt.Println("foo bar")
+		if false {
+			goto SOME_LABEL
+		}
+		if true {
+			goto OTHER_LABEL
+		}
+		break
+	SOME_LABEL:
+		fmt.Println("fizz")
+		break
+	OTHER_LABEL:
+		fmt.Println("buzz")
+		break // MATCH /useless break in case clause/
+		fmt.Println("foobar")
+		break // MATCH /useless break in case clause/
+	}
 }

--- a/testdata/useless_break.go
+++ b/testdata/useless_break.go
@@ -91,21 +91,12 @@ func UselessBreaks() {
 	// issue #1281
 	switch 1 {
 	case 1:
-		fmt.Println("foo bar")
-		if false {
-			goto SOME_LABEL
-		}
-		if true {
-			goto OTHER_LABEL
-		}
-		break
-	SOME_LABEL:
-		fmt.Println("fizz")
-		break
-	OTHER_LABEL:
-		fmt.Println("buzz")
+		fmt.Println("foo")
 		break // MATCH /useless break in case clause/
-		fmt.Println("foobar")
-		break // MATCH /useless break in case clause/
+	case 2:
+		fmt.Println("bar")
+		break 
+		fmt.Println("baz")
+	case 3:
 	}
 }

